src/rust/src/x509/verify.rs

```diff
@@ -43,6 +46,129 @@ impl CryptoOps for PyCryptoOps {
     }
 }
 
+struct PyCryptoPolicy<'a>(Policy<'a, PyCryptoOps>);
+
+/// This enum exists solely to provide heterogeneously typed ownership for `OwnedPolicy`.
+enum SubjectOwner {
+    DNSName(String),
+    IPAddress(pyo3::Py<pyo3::types::PyBytes>),
+}
+
+self_cell::self_cell!(
+    struct OwnedPolicy {
+        owner: SubjectOwner,
+
+        #[covariant]
+        dependent: PyCryptoPolicy,
+    }
+);
+
+#[pyo3::pyclass(
+    name = "ServerVerifier",
+    module = "cryptography.hazmat.bindings._rust.x509"
+)]
+struct PyServerVerifier {
+    // TODO: Switch this to `Py<PyString>` once Pyo3's `to_str()` preserves a
+    // lifetime relationship between an a `PyString` and its borrowed `&str`
+    // reference in all limited API builds. PyO3 can't currently do that in
+    // older limited API builds because it needs `PyUnicode_AsUTF8AndSize` to do
+    // so, which was only stabilized with 3.10.
+    py_subject: pyo3::Py<pyo3::PyAny>,
```

Comment:
You can use `pyo3(get, name = "subject")` here (https://pyo3.rs/v0.19.2/class#object-properties-using-pyo3get-set)

Reply:
[fbb2e6a](https://github.com/pyca/cryptography/pull/9405/commits/fbb2e6a63d521795acd4b38d285dac8a71b50626)